### PR TITLE
device-agnostic code

### PR DESCRIPTION
**Pull request by jin-s13**

### CODE REVIEW

File: mmpose/models/detectors/bottom_up.py [hunk 1]
```diff
@@ -194,7 +194,6 @@ def forward_test(self, img, img_metas, **kwargs):
         tags_list = []
         for idx, s in enumerate(sorted(test_scale_factor, reverse=True)):
             image_resized = aug_data[idx]
-            image_resized = image_resized.cuda(non_blocking=True)
```
Comment (innerlee) [inline mmpose/models/detectors/bottom_up.py:197]: is `image_resized` on gpu already?

File: tests/test_model/test_top_down_forward.py [hunk 2]
```diff
@@ -43,10 +43,11 @@ def test_topdown_forward():
     assert isinstance(losses, dict)
 
     # Test forward test
-    with torch.no_grad():
-        detector = detector.cuda()
-        _ = detector.forward(
-            imgs.cuda(), img_metas=img_metas, return_loss=False)
+    if torch.cuda.is_available():
```
Comment (innerlee) [inline tests/test_model/test_top_down_forward.py:46]: else?

Comment (innerlee) [reply tests/test_model/test_top_down_forward.py:46]: Tests that require cuda could be annotated with some kind of python decorators.

Comment (innerlee) [reply tests/test_model/test_top_down_forward.py:46]: btw why train does not need .cuda()

File: mmpose/models/losses/multi_loss_factory.py [hunk 3]
```diff
@@ -9,20 +9,19 @@
...
 
     Args:
         t (torch.Tensor): input
         requires_grad (bool): Option to use requires_grad.
-        need_cuda (bool): Opthin to use cuda.
+        device: torch device
     Returns:
         inp (torch.Tensor): zero input.
     """
     inp = torch.autograd.Variable(t, requires_grad=requires_grad)
```
Comment (innerlee) [inline mmpose/models/losses/multi_loss_factory.py:22]: Unrelated to this pr, but Variable has been deprecated. https://pytorch.org/docs/stable/autograd.html#variable-deprecated

Comment (innerlee) [reply mmpose/models/losses/multi_loss_factory.py:22]: Maybe we can remove all usage of Variables in a separate PR.

Comment (jin-s13) [reply mmpose/models/losses/multi_loss_factory.py:22]: Good idea.

**Pull request merged**